src/ncp-spinel/SpinelNCPInstance.cpp

```diff
@@ -4062,6 +4099,45 @@ SpinelNCPInstance::handle_ncp_spinel_value_is(spinel_prop_key_t key, const uint8
 			signal_property_changed(kWPANTUNDProperty_TmfProxyStream, data);
 		}
 
+	} else if (key == SPINEL_PROP_THREAD_UDP_PROXY_STREAM) {
+		const uint8_t* frame_ptr(NULL);
+		unsigned int frame_len(0);
+		uint16_t peer_port = 0;
+		in6_addr *peer_addr;
+		uint16_t sock_port = 0;
+		spinel_ssize_t ret;
+		Data data;
+
+		ret = spinel_datatype_unpack(
+			value_data_ptr,
+			value_data_len,
+			SPINEL_DATATYPE_DATA_S
+			SPINEL_DATATYPE_UINT16_S    // Peer port
+			SPINEL_DATATYPE_IPv6ADDR_S  // Peer address
+			SPINEL_DATATYPE_UINT16_S,   // Sock port
+			&frame_ptr,
+			&frame_len,
+			&peer_port,
+			&peer_addr,
+			&sock_port
+		);
+
+		__ASSERT_MACROS_check(ret > 0);
+
+		// Analyze the packet to determine if it should be dropped.
+		if ((ret > 0)) {
+			// append frame
+			data.append(frame_ptr, frame_len);
+			// pack the locator in big endian.
+			data.push_back(peer_port >> 8);
+			data.push_back(peer_port & 0xff);
+			data.append(peer_addr->s6_addr, sizeof(*peer_addr));
+			// pack the port in big endian.
+			data.push_back(sock_port >> 8);
+			data.push_back(sock_port & 0xff);
+			signal_property_changed(kWPANTUNDProperty_UdpProxyStream, data);
```

Comment:
How about providing a get handler for the `kWPANTUNDProperty_UdpProxyStream` property to give us the last received UDP stream (last one for which we have emitted `signal_property_changed`).
We can save the `data` from last received UDP stream here in some member variable and return it from the get-handler. This way users that only parse the property name (and not the value) from `signal_property_changed` would have a way to get the frame.

Thoughts?

Reply:
Do you have a specific use case in mind for this? Even though users only parse the property name first, they are still able to further parse the value, aren't they? And it seems we cannot guarantee the last received UDP stream is still the original one when users ask for that through another d-bus call.

Reply:
I was thinking of testing as the main use-case for this. 
I agree users should parse the emitted dbus signal to get both name and value so that they ensure that they don't miss any frames.

Reply:
Thanks for the clarification. How about we implement that when we actually use it? I think it's better to keep a PR small.

Reply:
Sounds good to me. :+1: